source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -0,0 +1,154 @@
+  {
+    namespace Interpolator
+    {
+      template <int dim>
+      std::vector<std::vector<double> >
+      BilinearLeastSquares<dim>::properties_at_points(const std::multimap<types::LevelInd, Particle<dim> > &particles,
+                                                      const std::vector<Point<dim> > &positions,
+                                                      const ComponentMask &selected_properties,
+                                                      const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell) const
+      {
+        AssertThrow(dim == 2,
+                    ExcMessage("Currently, the particle interpolator 'bilinear' is only supported for 2D models."));
+
+        typename parallel::distributed::Triangulation<dim>::active_cell_iterator found_cell;
+
+        Point<dim> approximated_cell_midpoint;
+
+        if (cell == typename parallel::distributed::Triangulation<dim>::active_cell_iterator())
+          {
+            // We can not simply use one of the points as input for find_active_cell_around_point
+            // because for vertices of mesh cells we might end up getting ghost_cells as return value
+            // instead of the local active cell. So make sure we are well in the inside of a cell.
+            Assert(positions.size() > 0,
+                   ExcMessage("The particle property interpolator was not given any "
+                              "positions to evaluate the particle cell_properties at."));
+
+            approximated_cell_midpoint = std::accumulate (positions.begin(), positions.end(), Point<dim>())
+                                         / static_cast<double> (positions.size());
+
+            found_cell =
+              (GridTools::find_active_cell_around_point<> (this->get_mapping(),
+                                                           this->get_triangulation(),
+                                                           approximated_cell_midpoint)).first;
+          }
+        else
+          found_cell = cell;
+
+        const types::LevelInd cell_index = std::make_pair<unsigned int, unsigned int> (found_cell->level(),found_cell->index());
+        const std::pair<typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator,
+              typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator> particle_range = particles.equal_range(cell_index);
+
+        const unsigned int n_particles = std::distance(particle_range.first,particle_range.second);
+        const unsigned int n_particle_properties = particles.begin()->second.get_properties().size();
+
+        std::vector<std::vector<double> > cell_properties(positions.size(),
+                                                          std::vector<double>(n_particle_properties, numbers::signaling_nan<double>()));
+
+        unsigned int property_index = 0;
+        for (unsigned int i=0; i < n_particle_properties; i++)
+          if (selected_properties[i])
```

Comment:
So this interpolator only works if only one property is selected in the component_mask? Please check that this is the case at the beginning of the function, currently you silently only compute the last selected component.